mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/MailboxListenerRegistryTest.java

```diff
@@ -235,5 +239,29 @@ void removeListenerOnlyRunTaskOnceForEmptyRegistry() throws Exception {
 
             assertThat(runIfEmptyCount.get()).isEqualTo(1);
         }
+
+        @Test
+        void iterationShouldPerformOnASnapshotOfListenersSet() throws Exception {
+            MailboxListener listener1 = event -> {};
+            MailboxListener listener2 = event -> {};
+            MailboxListener listener3 = event -> {};
+            MailboxListener listener4 = event -> {};
+            MailboxListener listener5 = event -> {};
+
+            testee.addListener(KEY_1, listener1, NOOP);
+            testee.addListener(KEY_1, listener2, NOOP);
+            testee.addListener(KEY_1, listener3, NOOP);
+            testee.addListener(KEY_1, listener4, NOOP);
+            testee.addListener(KEY_1, listener5, NOOP);
+
+            Mono<List<MailboxListener>> listeners = testee.getLocalMailboxListeners(KEY_1)
+                .publishOn(Schedulers.elastic())
+                .delayElements(Duration.ofMillis(100))
+                .collectList();
+
+            testee.removeListener(KEY_1, listener5, NOOP);
+
+            assertThat(listeners.block(Duration.ofSeconds(10))).hasSize(5);
```

Comment:
I miss something here: you add 5 listeners, you remove 1, and you expect still 5?

Reply:
Yes, of course, any changed to the listeners list done once an iteration is already started should not affect this iteration

Reply:
thanks, this asynchronous but triggered by publish thing will always bother me :)